### ISSUE TITLE
L: Week 7 - Migrate changes from `feat/eduai-redesign` to `cleanup/pr-583-final-state`, prep for merge

### ISSUE THREAD

**Issue by yta3216** [EduAI-Lab/EduAI#653]
Fully resolve all issues with the new `cleanup/pr-583-final-state` branch that @Whiteknight07 created to align `feat/eduai-redesign`'s git history with `development` (thank you Stavan). Ensure that migrating to this branch is clean and that every feature present in `feat/eduai-redesign` is in the new branch.

Resolved issues while preparing branch for merge:
AI Tutor:
- [x] Instructors cannot publish/unpublish a course
- [x] Unit Admins cannot import courses from EduAI
- [x] Unit admins will crash the entire app if they click on any menu item in the course detail view
- [x] Students show up as their ID (e.g. `seed_user_student_02`, `seed_user_student_04`) instead of their actual names
- [x] Unit admins cannot create or modify courses
- [x] Students and TAs cannot view the courses they are enrolled in/assigned to
- [x] Unit admins cannot view courses
- [x] Chat history is stored in localStorage and will be lost when starting on a new device

EduAI:
- [x] Students are not automatically seeded with their student ID
- [x] Core is no ready to share its course data with the other two extensions. Prep these so that Ehsan and Ahab can work off a working endpoint.
- [x] Long lists of units will cause the unit modal to overflow when assigning unit admins

QM:
- [x] Course access is completely broken. Every course is visible to every user but no one can interact with them.
- [x] Courses are completely missing any form of role guards in the backend

All:
- [x] E2E tests are broken on all platforms. Update tests to match the new UI and ensure they all pass.

EPIC: #62 
Hours to complete (Week 7) [Ye]: 9 hours
